web/src/components/product/ProductPage.test.jsx

```diff
@@ -0,0 +1,382 @@
+    onStatusChange: jest.fn()
+  };
+
+  mockSoftware = {
+    probe: jest.fn(),
+    getStatus: jest.fn().mockResolvedValue(),
+    onStatusChange: jest.fn(),
+    product: {
+      getSelected: selectedProduct.id,
+      select: jest.fn().mockResolvedValue(),
+      onChange: jest.fn()
+    }
+  };
+
+  mockRegistration = {
+    requirement: "not-required",
+    code: null,
+    email: null
+  };
+
+  createClient.mockImplementation(() => (
+    {
+      manager: mockManager,
+      software: mockSoftware
+    }
+  ));
+});
+
+it("renders the product name and description", async () => {
+  installerRender(<ProductPage />);
+  await screen.findByText("Test Product1");
+  await screen.findByText("Test Product1 description");
+});
+
+describe("if the product is already registered", () => {
+  beforeEach(() => {
+    mockRegistration.code = "111222";
+    mockRegistration.email = "test@test.com";
+  });
+
+  it("shows the information about the registration", async () => {
+    installerRender(<ProductPage />);
+    await screen.findByText("111222");
+    await screen.findByText("test@test.com");
+  });
+});
+
+describe("if the product does not require registration", () => {
+  beforeEach(() => {
+    mockRegistration.requirement = "not-required";
+  });
+
+  it("shows a button to change the product", async () => {
+    installerRender(<ProductPage />);
+    await screen.findByRole("button", { name: "Change product" });
+  });
+
+  it("does not show a button to register the product", async () => {
+    installerRender(<ProductPage />);
+    expect(screen.queryByRole("button", { name: "Register" })).not.toBeInTheDocument();
+  });
+});
+
+describe("if the product requires registration", () => {
+  beforeEach(() => {
+    mockRegistration.requirement = "required";
+  });
+
+  it("shows a button to change the product", async () => {
+    installerRender(<ProductPage />);
+    await screen.findByRole("button", { name: "Change product" });
+  });
+
+  describe("and the product is not registered yet", () => {
+    beforeEach(() => {
+      mockRegistration.code = null;
+    });
+
+    it("shows a button to register the product", async () => {
+      installerRender(<ProductPage />);
+      await screen.findByRole("button", { name: "Register" });
+    });
+  });
+
+  describe("and the product is already registered", () => {
+    beforeEach(() => {
+      mockRegistration.code = "11112222";
+    });
+
+    it("shows a button to deregister the product", async () => {
+      installerRender(<ProductPage />);
+      await screen.findByRole("button", { name: "Deregister" });
+    });
+  });
+});
+
+describe("when the services are busy", () => {
+  beforeEach(() => {
+    mockRegistration.requirement = "required";
+    mockRegistration.code = null;
+    mockSoftware.getStatus = jest.fn().mockResolvedValue(BUSY);
+  });
+
+  it("shows disabled buttons", async () => {
+    await act(async () => installerRender(<ProductPage />));
```

Comment:
Question: Why do you need to use `act` here?

Reply:
Fixed.